packages/edit-site/src/store/actions.js

```diff
@@ -141,11 +141,18 @@ export const removeTemplate =
 				throw lastError;
 			}
 
+			// Depending on how the entity was retrieved it's title might be
+			// an object or simple string.
+			const templateTitle =
+				typeof template.title === 'string'
+					? template.title
+					: template.title?.rendered;
```

Comment:
I put `?` just in case. This implementation is also done in the following locations:

https://github.com/WordPress/gutenberg/blob/3fd16c019bee2e0bfa2769c1d68907aba0d482c1/packages/editor/src/store/selectors.js#L1611